code/datalab-app/src/app/components/notebooks/NotebooksContainer.js

```diff
@@ -4,17 +4,36 @@ import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import notebookActions from '../../actions/notebookActions';
 import NotebookButton from './NotebookButton';
+import { JUPYTER, ZEPPELIN } from '../../../shared/notebookTypes';
 
 class NotebooksContainer extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.openNotebook = this.openNotebook.bind(this);
+  }
   componentWillMount() {
     this.props.actions.loadNotebooks();
   }
 
+  openNotebook(notebook) {
```

Comment:
The code in this function doesn't feel like it belongs in the container.  Could it maybe be pushed down into the actions?